File: net/sync/service-interface/src/test/java/de/danoeh/antennapod/net/sync/serviceinterface/SynchronizationProviderTest.java
package de.danoeh.antennapod.net.sync.serviceinterface;

import static org.junit.Assert.assertArrayEquals;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

import org.junit.Test;

public class SynchronizationProviderTest {

    @Test
    public void fromIdentifierWithNullReturnsNull() {
        assertNull(SynchronizationProvider.fromIdentifier(null));
    }

    @Test
    public void fromIdentifierWithEmptyStringReturnsNull() {
        assertNull(SynchronizationProvider.fromIdentifier(""));
    }

    @Test
    public void fromIdentifierWithUnknownStringReturnsNull() {
        assertNull(SynchronizationProvider.fromIdentifier("NOT_A_PROVIDER"));
    }

    @Test
    public void fromIdentifierIsCaseSensitiveAndReturnsNullForLowerCase() {
        assertNull(SynchronizationProvider.fromIdentifier("gpodder_net"));
    }

    @Test
    public void fromIdentifierWithExactMatchReturnsGpodderNet() {
        assertEquals(SynchronizationProvider.GPODDER_NET, SynchronizationProvider.fromIdentifier("GPODDER_NET"));
    }

    @Test
    public void fromIdentifierWithExactMatchReturnsNextcloudGpodder() {
        assertEquals(SynchronizationProvider.NEXTCLOUD_GPODDER,
                SynchronizationProvider.fromIdentifier("NEXTCLOUD_GPODDER"));
    }

    @Test
    public void getIdentifierReturnsExactPersistedStringForGpodderNet() {
        assertEquals("GPODDER_NET", SynchronizationProvider.GPODDER_NET.getIdentifier());
    }

    @Test
    public void getIdentifierReturnsExactPersistedStringForNextcloudGpodder() {
        assertEquals("NEXTCLOUD_GPODDER", SynchronizationProvider.NEXTCLOUD_GPODDER.getIdentifier());
    }

    @Test
    public void valuesReturnsBothConstantsInDeclarationOrder() {
        assertArrayEquals(
                new SynchronizationProvider[] {
                    SynchronizationProvider.GPODDER_NET, SynchronizationProvider.NEXTCLOUD_GPODDER
                },
                SynchronizationProvider.values());
    }
}

File: net/sync/service-interface/src/test/java/de/danoeh/antennapod/net/sync/serviceinterface/SynchronizationProviderTest.kt
package de.danoeh.antennapod.net.sync.serviceinterface

import org.junit.Assert.assertArrayEquals
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Test

class SynchronizationProviderTest {

    @Test
    fun fromIdentifierWithNullReturnsNull() {
        assertNull(SynchronizationProvider.fromIdentifier(null))
    }

    @Test
    fun fromIdentifierWithEmptyStringReturnsNull() {
        assertNull(SynchronizationProvider.fromIdentifier(""))
    }

    @Test
    fun fromIdentifierWithUnknownStringReturnsNull() {
        assertNull(SynchronizationProvider.fromIdentifier("NOT_A_PROVIDER"))
    }

    @Test
    fun fromIdentifierIsCaseSensitiveAndReturnsNullForLowerCase() {
        assertNull(SynchronizationProvider.fromIdentifier("gpodder_net"))
    }

    @Test
    fun fromIdentifierWithExactMatchReturnsGpodderNet() {
        assertEquals(SynchronizationProvider.GPODDER_NET, SynchronizationProvider.fromIdentifier("GPODDER_NET"))
    }

    @Test
    fun fromIdentifierWithExactMatchReturnsNextcloudGpodder() {
        assertEquals(
            SynchronizationProvider.NEXTCLOUD_GPODDER,
            SynchronizationProvider.fromIdentifier("NEXTCLOUD_GPODDER")
        )
    }

    @Test
    fun getIdentifierReturnsExactPersistedStringForGpodderNet() {
        assertEquals("GPODDER_NET", SynchronizationProvider.GPODDER_NET.identifier)
    }

    @Test
    fun getIdentifierReturnsExactPersistedStringForNextcloudGpodder() {
        assertEquals("NEXTCLOUD_GPODDER", SynchronizationProvider.NEXTCLOUD_GPODDER.identifier)
    }

    @Test
    fun valuesReturnsBothConstantsInDeclarationOrder() {
        assertArrayEquals(
            arrayOf(SynchronizationProvider.GPODDER_NET, SynchronizationProvider.NEXTCLOUD_GPODDER),
            SynchronizationProvider.values()
        )
    }
}
